Sources/Plasma/CoreLib/hsThread_Unix.cpp

```diff
@@ -81,8 +81,14 @@ hsGlobalSemaphore::hsGlobalSemaphore(int initialValue, const ST::string& name)
     fPSema = nullptr;
     fNamed = !name.empty();
     if (fNamed) {
+        const char* semName = name.c_str();
+        if (semName[0] != '/') {
+            /* UNIX named semaphores need to start with a slash */
+            semName = ST::format("/{}", name).c_str();
+        }
```

Comment:
To avoid using a dangling pointer (the string produced by `ST::format` goes out of scope)
```suggestion
        ST::string semName = name;
        if (semName.front() != '/') {
            /* UNIX named semaphores need to start with a slash */
            semName = ST::format("/{}", name);
        }
```